### ISSUE TITLE
Further selection refinements for collapsed selectlist

### ISSUE THREAD

**Issue by scottaohara** [openui/open-ui#897]
This is a follow on to #855

When reviewing [this selectlist demo](https://jsfiddle.net/xhsofq2z/4/) I noticed that while the behavior was changed to no longer have selection follow focus - and thus behave more similarly to macOS, that even the macOS-like behavior had one odd use case.  When the selectlist is in the collapsed state, I can start typing characters and that will cause a selection change to the option that best matches the characters I typed. That automatic change _does not occur_ when the listbox is rendered.   This is, however, how the macOS select element behaves currently (though I can briefly pause and start typing new characters and the selected option will change to match that new set of characters - but the selectlist demo only seems to work once in this regard.  If i want to start typing to change options, i have to move focus away from the selectlist button and then move focus back for typing to change the option once again.)

It seems to me that we should probably change this behavior for selectlist, as for one it already seems strange to me that this even happens with macOS's select element - since all other interactions with the button part open the listbox, rather than immediately change the selected option.  But on Windows, where the behavior is shifting away from current expectations to align more with macOS - this is the only behavior that would remain as it does with the current select element - and it honestly feels like a bug / a missed behavioral change (e.g., start typing characters and that opens the listbox / moves to the matching option).

So, tldr;
since it was previously resolved to not be so eager with selection for selectlist, i'd submit character typing when the selectlist's listbox is in the collapsed state, and having that autoselect an option / causing a change event, is a lingering eagerness that should be changed as well.

**Comment by josepharhar** [comment]
So we shouldn't do typeahead while the listbox is closed? Seems fine to me I guess. I suppose some people might like this behavior, but i shouldn't be too hard to also press the spacebar first to open the listbox right?

**Comment by mfreed7** [comment]
Would it be out of line for typeahead to both a) open the listbox, and b) start matching the typeahead? That would at least align expectations that events are fired as the selection is changed because there's an open picker.

**Comment by josepharhar** [comment]
Yeah I feel that we should do something since typeahead when the listbox is closed seems to work on all platforms for the existing `<select>` element
